### PR TITLE
[Feature]: Video demo

### PR DESCRIPTION
**Pull request by steven0129**
Demo with inference on video

### CODE REVIEW

**Review by jobliw — APPROVED** [review]

**Comment by xvjiarui** [comment]
@steven0129 
Thx for your contribution and sorry for the late review.
1. Could you please fix the lint error?
2. We may make this script accept used arguments, like `image_demo.py`.

**Comment by steven0129** [comment]
Thank's for your review. Okay, I'll try to fix these error.

File: demo/video_demo.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        if ret:
+            result = inference_detector(model, frame)
+            frame = model.show_result(frame, result, score_thr=args.score_thr)
+            if args.show:
+                cv2.namedWindow('video', 0)
+                mmcv.imshow(frame, 'video', args.wait_time)
+            if video_writer:
+                video_writer.write(frame)
+        else:
+            print(f'Fail to read {args.video} video')
```
Comment (ZwwWayne) [inline demo/video_demo.py:64]: Fail to read video from {}

File: demo/video_demo.py [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+def main():
+    args = parse_args()
+    assert args.out or args.show, \
+        ('Please specify at least one operation (save/show the '
+         'video) with the argument "--out" or "--show"')
+
+    model = init_detector(args.config, args.checkpoint, device=args.device)
+
+    video_capture = cv2.VideoCapture(args.video)
```
Comment (ZwwWayne) [inline demo/video_demo.py:37]: We should try to use the video API in MMCV, see [here](https://github.com/open-mmlab/mmcv/blob/master/mmcv/video/io.py#L42).

File: demo/video_demo.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    if args.out:
+        fourcc = cv2.VideoWriter_fourcc(*'mp4v')
+        video_writer = cv2.VideoWriter(
+            args.out, fourcc, video_reader.fps,
+            (video_reader.width, video_reader.height))
+    else:
+        video_writer = None
+
+    prog_bar = mmcv.ProgressBar(len(video_reader))
+    for frame in video_reader:
```
Comment (ZwwWayne) [inline demo/video_demo.py:47]: for frame in mmcv.track_iter_progress(video_reader):

File: demo/video_demo.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        prog_bar.update()
+        result = inference_detector(model, frame)
+        frame = model.show_result(frame, result, score_thr=args.score_thr)
+        if args.show:
+            cv2.namedWindow('video', 0)
+            mmcv.imshow(frame, 'video', args.wait_time)
+        if video_writer:
```
Comment (ZwwWayne) [inline demo/video_demo.py:54]: Using args.out maybe more clear

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**